File: microservices/service-vendeur/src/main/java/cnam/liban/jalal9506f/MagasinEnLigne/models/Vendeur.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cnam.liban.jalal9506f.MagasinEnLigne.models;

import java.util.UUID;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

/**
 *
 * @author jalal
 */
@Document
public class Vendeur {

    @Id
    @Field("id")
    private UUID id;
    @Field("fullName")
    private String fullName;
    @Field("password")
    private String password;
    @Field("phoneNumber")
    private String phoneNumber;
    @Field("email")
    private String email;

    public Vendeur() {
        this.id = UUID.randomUUID();
    }

    public Vendeur(UUID id, String fullName, String password, String phoneNumber) {
        this.id = id;
        this.fullName = fullName;
        this.password = password;
        this.phoneNumber = phoneNumber;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: android/app/src/main/java/com/cnam/magasinenligne/api/models/Vendeur.kt
package com.cnam.magasinenligne.api.models

import com.google.gson.annotations.SerializedName

data class Vendeur(
    @SerializedName("id") val id: String = "",
    @SerializedName("fullName") val fullName: String = "",
    @SerializedName("password") val password: String = "",
    @SerializedName("phoneNumber") val phoneNumber: String = "",
    @SerializedName("email") val email: String? = "",
    @SerializedName("image") val image: String? = null
)

data class SingleVendeurResponse(
    @SerializedName("Response") val vendeur: Vendeur
) : CommonResponse()

data class MultipleVendeurResponse(
    @SerializedName("Response") val vendeurs: List<Vendeur>
) : CommonResponse()